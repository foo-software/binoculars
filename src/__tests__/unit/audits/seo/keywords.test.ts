import ArtifactsInterface from '../../../../interfaces/Artifacts';
import KeywordsAudit from '../../../../audits/seo/keywords';

describe('KeywordsAudit', () => {
  describe('KeywordsAudit.meta', () => {
    it('should match snapshot', () => {
      expect(KeywordsAudit.meta).toMatchSnapshot();
    });
  });

  describe('KeywordsAudit.audit', () => {
    let artifacts: ArtifactsInterface = {};

    beforeAll(() => {
      artifacts = {};
    });

    it('should return a passing score when sufficient keywords exist between title, description and content', () => {
      artifacts = {
        Elements: {
          body: [
            {
              text:
                'my page has meaningful content, yes it has MEANINGFUL CONTENT',
            },
          ],
        },
        MetaElements: [
          {
            name: 'description',
            content: 'description: some meaningful content',
          },
        ],
        TitleElement: {
          text: 'title: some meaningful content',
        },
      };
      const result = KeywordsAudit.audit(artifacts);
      expect(result.score).toEqual(1);
    });

    it('should return a failing score when less than 2 keywords exist in the content', () => {
      artifacts = {
        Elements: {
          body: [
            {
              text: 'my page has content, yes it has CONTENT',
            },
          ],
        },
        MetaElements: [
          {
            name: 'description',
            content: 'description: some meaningful content',
          },
        ],
        TitleElement: {
          text: 'title: some meaningful content',
        },
      };
      const result = KeywordsAudit.audit(artifacts);
      expect(result.score).toEqual(0);
    });

    it('should return a failing score when less than 2 keywords exist in the description', () => {
      artifacts = {
        Elements: {
          body: [
            {
              text:
                'my page has meaningful content, yes it has MEANINGFUL CONTENT',
            },
          ],
        },
        MetaElements: [
          {
            name: 'description',
            content: 'description: some content',
          },
        ],
        TitleElement: {
          text: 'title: some meaningful content',
        },
      };
      const result = KeywordsAudit.audit(artifacts);
      expect(result.score).toEqual(0);
    });

    it('should return a failing score when less than 2 keyword instances exist in the content', () => {
      artifacts = {
        Elements: {
          body: [
            {
              text: 'my page has meaningful content',
            },
          ],
        },
        MetaElements: [
          {
            name: 'description',
            content: 'description: some meaningful content',
          },
        ],
        TitleElement: {
          text: 'title: some meaningful content',
        },
      };
      const result = KeywordsAudit.audit(artifacts);
      expect(result.score).toEqual(0);
    });
  });
});
