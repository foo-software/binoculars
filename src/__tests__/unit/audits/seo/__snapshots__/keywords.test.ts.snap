// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KeywordsAudit KeywordsAudit.meta should match snapshot 1`] = `
Object {
  "description": "Keywords present in the title, description and content of a page help narrow targeted search result ranking. At least 2 words should exist in the title, description and exist at least 2 times in the content of the page",
  "failureTitle": "Keywords of at least 2 words do not exist in the title, description and exist at least 2 times in the content of the page",
  "id": "keywords",
  "requiredArtifacts": Array [
    "Elements",
    "MetaElements",
    "TitleElement",
  ],
  "title": "Has keyword content",
}
`;
